backend/src/Parser/Parser.fs

```diff
@@ -473,7 +508,20 @@ let parseTestFile (filename : string) : Module =
     checker.ParseFile(fsharpFilename, Text.SourceText.ofString input, parsingOptions)
     |> Async.RunSynchronously
 
-  let rec convertType (typ : SynType) : PT.DType =
+  // TODO: support non-0 versions of user-defined types
+  let matchingUserTypes
```

Comment:
I don't think we should overlap parsing and type matching like this. It's weird if you change the name of your type definition and then the parser stops working.

Reply:
It would be better if parsing produced the same output regardless of the type in existence.